### ISSUE TITLE
Documentation generation problem

### ISSUE THREAD

**Issue by cognitivegears** [NuSkooler/enigma-bbs#405]
**Describe the Bug**

It looks like post-merge  the main documentation isn't looking right unfortunately. From what I can tell, I believe it's trying to use a version built by github-pages still instead of the action. The page may need to be switched over in the settings for the repo to use the gh-pages branch I believe.

**Comment by NuSkooler** [comment]
I'd like to keep it on `master` since that's where the action will be. Looks like the older setup and the new one are both running according to Actions.  Currently set to `master` and `/docs`.

**Comment by cognitivegears** [comment]
In this case this is the branch that holds the results of the compilation, rather than the source branch as it was before. If you look at what is in the gh-pages branch it contains the compilation results from Jekyll. This can't be set to master because then the output would overwrite the actual source code. 

The reason that master worked before was because GitHub picked up the branch and did the compilation itself, but since it uses an old version of Jekyll it doesn't have all the collection features we needed. So instead, the new code does the compilation using the Action so it builds with the latest Jekyll.

Overall the process of merging code etc remains the same, it's just the compilation process changes a bit:

Old:
1) Code merged into `master`
2) GitHub took the source from `master`, compiled it and deployed to the static site

New:
1) Code merged into `master`
2) GitHub Action kicks off, compiles the source from `master` and puts the results in `gh-pages`
3) GitHub takes the results from `gh-pages` and deploys to the static site.

In order to accommodate this change however the pages branch needs to be set to `gh-pages` in order to pick up this compiled result. (In the repository Settings, under GitHub Pages section, "gh-pages" should then be set as the source branch.) 

I apologize I should have explained that better as part of the PR itself, please let me know if you have any questions or if there is anything I can help with.

**Comment by NuSkooler** [comment]
@cognitivegears Thanks for the info, probably something I could have just RTFM'd, so I appreciate you taking the time to spell it out :)

I've updated to point at the new branch. Should be good to go now I believe.

**Comment by cognitivegears** [comment]
No worries at all I appreciate the help. Unfortunately still hasn't corrected, but if that's been changed than I'm pretty sure that is just because nothing else has been checked into master to re-execute the actions yet. We can either check something else in, or if you can go into the Action tab and rerun the deploy. Really should be the last thing, I'm sorry about all the extra stuff on this one. If it's any consolation I'm working on some other stuff that will hopefully be good  (and less trouble! 🙂)

**Comment by NuSkooler** [comment]
@cognitivegears I did some things - did they work? :D

**Comment by cognitivegears** [comment]
I'm so sorry, life got in the way and I was just able to get back to this. Unfortunately, it did not work. But, I did get the exact steps to do that I think can get it resolved, see below:

1) First, click the Settings button to go into the repository settings

![1_click_settings](https://user-images.githubusercontent.com/3256153/153716982-a5594586-de18-4045-9b8c-730a9aff495c.png)

--------------------------

2) On the left hand side of the screen, select Pages settings section

![2_select_Pages_settings](https://user-images.githubusercontent.com/3256153/153716997-fb828889-629c-4772-9666-3c1aff9c2ee9.png)

--------------------------

3) Make sure that the branch for publishing is set to gh_pages, or set it to gh_pages if it is on something else.

![3_select_gh_pages_branch](https://user-images.githubusercontent.com/3256153/153717009-82342557-4c27-4055-8c9f-2b196932b370.png)

--------------------------

4) Make sure deployment folder is set to root

![4_make_sure_folder_is_set_to_root](https://user-images.githubusercontent.com/3256153/153717042-d1f1c0e2-1995-4e50-a9fe-ac646fc45440.png)

_Note_: this is a bit counter-intuitive, since the files are in docs, but in our new github action it is publishing the output of the documentation to root - you can confirm that by looking at the files in gh_pages: https://github.com/NuSkooler/enigma-bbs/tree/gh-pages)

--------------------------
5) Click Save if you made any changes

![5_click_save](https://user-images.githubusercontent.com/3256153/153717137-a81c2f8c-71a3-4411-b847-d2bad8f97328.png)

--------------------------

6) Select the Actions tab at the top

![6_select_Actions](https://user-images.githubusercontent.com/3256153/153717145-ce40dd76-43fd-416a-a057-cb8d26c1277f.png)

--------------------------

7) Select "pages build and deployment"

![7_select_pages_build_and_deployment](https://user-images.githubusercontent.com/3256153/153717248-551eaaaa-d36e-4fec-bdb6-bb9983c325e1.png)
_Note_ - from the icon it failed last time, with the following error: "Error: No uploaded artifact was found! Please check if there are any errors at build step." But, I definitely see the output in the gh_pages branch, so that isn't the problem. Because of that I'm pretty sure that the settings either has it is set to another branch or the folder is set to docs.

--------------------------

8) Select the "..." menu

![8_select_dot_dot_dot](https://user-images.githubusercontent.com/3256153/153717345-13876098-2c49-4bd2-8af6-3a88453df5e7.png)

--------------------------
9) Click to Re-run all jobs

![9_select_re_run_all_jobs](https://user-images.githubusercontent.com/3256153/153717355-53551149-6c81-4a96-9649-b5ceb5122fae.png)

--------------------------

10) Hope it works.

Curse me for ever making contributions to your project :smile:. Seriously though, I'm so sorry this has taken a few tries, I'm guessing that it was because I hadn't mentioned the docs folder earlier - because when I made my fork the default is the root, so I didn't think about that you had to change it. In any case, sorry again and I hope this works, and I'll hopefully make it up to you with some more PRs that will hopefully go a lot smoother than this one.

**Comment by NuSkooler** [comment]
@cognitivegears Oh I understand, that's why I don't have much time to play with this stuff either currently. Thanks for the ELI5! I had it set to "/docs" still, so I think the "/root" should do the trick then. Just setting and saving it kicked of the job re-run again.

**Comment by cognitivegears** [comment]
Yay it worked! Thanks so much for your patience with this, we should be good going forward and we can probably close this issue. Thanks again.
